### PR TITLE
Update Dockerfile

### PR DESCRIPTION
**Pull request by sujithq**
ENTRYPOINT was missing

### CODE REVIEW

**Comment by im2nguyen** [comment]
Hey @sujithq, can you build the Dockerfile locally and run it? Once it works, I'll test it on my end and we can merge 😄

**Comment by im2nguyen** [comment]
Oh, completely forgot! 

The Dockerfile needs to build the UI first, since the `go build .` packages the `ui/dist` file into the binary

**Comment by sujithq** [comment]
I will take a look at this. I suppose it has to do with the embed thing.

**Comment by sujithq** [comment]
These are the steps to test:

1. pull this branch
2.  `cd ui`
2. delete the dist folder if it still exists
3. `cd ..`
4. Build docker `docker build . -t im2nguyen/rover --no-cache`
5. `cd examples\random-test`
6. Run one of thes commands
- `docker run --rm -it -p 9000:9000 -v $(pwd):/src im2nguyen/rover -workingDir src -planPath "/src/plan.out"`
- `docker run --rm -it -p 9000:9000 -v $(pwd):/src im2nguyen/rover -workingDir src -planPath "/src/plan.out" -standalone true`
- `docker run --rm -it -p 9000:9000 -v $(pwd):/src im2nguyen/rover -workingDir src"`
- `docker run --rm -it -p 9000:9000 -v $(pwd):/src im2nguyen/rover -workingDir src -standalone true`

ps: you need to provide -workingDir **src** in conjunction with you volume mapped to **src**

**Comment by im2nguyen** [comment]
Nice, I'll test it in the morning! Thank you for doing this!

It might be a good idea to add the build process in the GH actions for a new release, that way, it'll automatically build and published the latest version in `main` into DockerHub

**Comment by sujithq** [comment]
> Nice, I'll test it in the morning! Thank you for doing this!
> 
> It might be a good idea to add the build process in the GH actions for a new release, that way, it'll automatically build and published the latest version in `main` into DockerHub

see https://github.com/im2nguyen/rover/pull/41 for the push to docker hub. The Dockerfile should work as well.

File: Dockerfile [hunk 1]
```diff
@@ -1,12 +1,50 @@
...
+COPY ./ui/public ./public
+COPY ./ui/src ./src
+# build (to dist folder)
+RUN npm run build
 
+# Build rover
+FROM golang:1.17 AS rover
 WORKDIR /src
+# copy full source
+# COPY . .
```
Comment (im2nguyen) [inline Dockerfile:27]: is there a reason why we don't do `COPY . .`?

File: Dockerfile [hunk 2]
```diff
@@ -1,12 +1,50 @@
...
+RUN CGO_ENABLED=0 GOOS=linux go build -o rover .
 
-ENTRYPOINT [ "/bin/rover" ]
+# Release stage
+FROM hashicorp/terraform:$TF_VERSION AS release
+# copy terraform binary to the rover's default terraform path
+RUN cp /bin/terraform /usr/local/bin/terraform
+# copy rover binary
+COPY --from=rover /src/rover /bin/rover
+RUN chmod +x /bin/rover
```
Comment (im2nguyen) [inline Dockerfile:49]: This way, users don't need to specific `/src` on every single run 

```suggestion
RUN chmod +x /bin/rover

WORKDIR /src

```

**Review by im2nguyen — APPROVED** [review]

**Pull request merged**